data/provider/src/main/java/tech/pegasys/teku/api/NetworkDataProvider.java

```diff
@@ -89,4 +94,23 @@ public int getListenPort() {
   public Metadata getMetadata() {
     return new Metadata(network.getMetadata());
   }
+
+  public List<tech.pegasys.teku.api.response.v1.node.Peer> getPeers() {
+    return network.streamPeers().map(this::toPeer).collect(Collectors.toList());
+  }
+
+  public Optional<tech.pegasys.teku.api.response.v1.node.Peer> getPeerById(final String peerId) {
+    final NodeId nodeId = new LibP2PNodeId(PeerId.fromBase58(peerId));
```

Comment:
This is definitely not ideal.  It probably should ask the network to convert the `peerId` into the right type rather than hard coding a `LibP2PNodeId`.  Probably the simplest thing would be to overload `P2PNetwork.getPeer` to have a version that takes a string - then the conversion to a `NodeId` can be done internally.

Should consider what to do when the peerId isn't valid base58 - probably should return not found.

Reply:
wasn't super keen on that, means pulling in libp2p too, but wasn't sure what else to do.. will have a go with what you've suggested... otherwise i could potentially implement the interface for testing i guess...

Reply:
There's a `MockNodeId` which implements the interface for you.  I guess you could have another implementation that's a `SimpleNodeId` which just holds the base58 encoded node ID, but I think I'd be more inclined to leave it to the network to decode.